crates/subspace-service/src/sync_from_dsn/snap_sync.rs

```diff
@@ -80,6 +109,12 @@ pub(crate) async fn snap_sync<Block, AS, Client, PG>(
             Ok(()) => {
                 debug!("Snap sync finished successfully");
             }
+            // A fatal snap sync error requiring user intervention. The caller will log this error,
+            // so just return it to terminate the task.
+            error @ Err(Error::SnapSyncImpossible(_)) => {
+                return error;
+            }
+            // Other errors are non-fatal
             Err(error) => {
                 error!(%error, "Snap sync failed");
             }
```

Comment:
NIT: This certainly works, but I'd prefer a single `Err()` branch with `if matches!(error, Error::SnapSyncImpossible(_))` in it instead